test/functional/test_runner.py

```diff
@@ -322,6 +319,16 @@ def main():
         print("Rerun ./configure with --enable-wallet, --with-utils and --with-daemon and then make")
         sys.exit(0)
 
+    if not (len(ALL_SCRIPTS) == dict.fromkeys(ALL_SCRIPTS).keys()):
+        print("{}WARNING!{} There are duplicate tests in the test list:".format(BOLD[1], BOLD[0]))
+        duplicates = set()
+        for test in ALL_SCRIPTS:
+            if ALL_SCRIPTS.count(test) > 1:
+                duplicates.add(test)
+
+        for dup in duplicates:
+            print("- ", dup)
```

Comment:
WDYS if we exit with 1 in this case? Because locally we usually don't run all tests (we execute the individual one) and if the warning appears on Travis but the built passes, we won't notice it. Exiting with the error will ensure we won't have duplicate tests on master.

Reply:
>  Because locally we usually don't run all tests

I for one actually do run the whole test suite to see that I did not break anything else, and I do run it through the `test_runner`.

Either way I am also in favor of exiting with a non-zero exit code here.

Reply:
```python
 if os.getenv('TRAVIS') == 'true':
     # On travis this warning is an error to prevent merging duplicate tests on master
      sys.exit(1)
```
Will now kill it on travis but let it run locally.

Reply:
I for one am not comfortable with the idea of having tools that behave differently in the CI than they do locally. This is close to the Volkswagen Tests ;-)

If it dies locally - what's the problem? You go, fix the duplicate entry (like you should), and execute it.

Reply:
@scravy Yeah I think I can agree with this.

Reply:
nit: Then, it's no longer a WARNING but ERROR.

Reply:
I'd simply error locally too. If an error happens, it's good to fix it right away. I don't see a use case why we would like to keep the warning and have duplicate tests running on master.
+ it's easy to verify that it's correct, no need to manually create a failed build just to see it works (I didn't check about this ENV on Travis but believe it's there)